### PR TITLE
Fix `orientation` batching

### PR DESCRIPTION
**Pull request by amorehead**
* Fixes `orientation` feature batching for vector-equivariant methods.
* Without this fix, for vector-equivariant methods that use such features, precisely `batch.num_graphs - 2` nodes in each of their batched protein graphs would have their vector node input features normalized to a nonsense (invalid) vector.

### CODE REVIEW

File: proteinworkshop/features/node_features.py [hunk 1]
```diff
@@ -153,12 +153,17 @@ def compute_surface_feat(
...
 ) -> OrientationTensor:
     if X.ndim == 3:
         X = X[:, ca_idx, :]
-    forward = _normalize(X[1:] - X[:-1])
-    backward = _normalize(X[:-1] - X[1:])
-    forward = F.pad(forward, [0, 0, 0, 1])
-    backward = F.pad(backward, [0, 0, 1, 0])
-    return torch.cat((forward.unsqueeze(-2), backward.unsqueeze(-2)), dim=-2)
+    batched_orientations = []
+    for batch_id in batch_index.unique(sorted=True):
```
Comment (a-r-j) [inline proteinworkshop/features/node_features.py:161]: I think this can be vectorised by using indices from `batch._slice_dict["coords"] to set the relevant indices to 0

Comment (amorehead) [reply proteinworkshop/features/node_features.py]: OK. Have a look at this new (slightly longer) vectorized implementation: https://github.com/a-r-j/ProteinWorkshop/pull/58/commits/a1d864bb00d5538c99ae42bfcb5595b34e1a5e47

Comment (amorehead) [reply proteinworkshop/features/node_features.py]: Truthfully, all of the asserts (commented out for now) should probably be made into a unit test to detect issues like this in the future.

**Comment by a-r-j** [comment]
@amorehead Happy for me to merge?

**Comment by amorehead** [comment]
@arian, yep, should be good to go. Let me know if you run into any unexpected edge cases while training. I believe this should handle all of them.
